src/Containers/NutritionFact/NutritionFact.tsx

```diff
@@ -0,0 +1,307 @@
+import React,{useCallback,useMemo,useRef,forwardRef,useEffect,useState} from 'react'
+import styled from 'styled-components'
+import { Mixins } from '../../Utils'
+import {
+    Main,
+    MainInterface,
+} from '../../Utils/BaseStyles';
+import Theme from '../../Themes/ThemeTemplate'
+import {Input} from '../../Inputs/Input/Input'
+
+const EXTRA_PIXEL=1
+const ENTRY_INPUT_WIDTH=20
+const HEADER_ENTRY_INPUT_WIDTH=40
+
+interface IAdditionalProps{
+    /* if the heading entry must to be taken as the reference entry for taking width from */
+    ref?:boolean;
+    /* if the content of the heading entry must be delayed after setting max width of the main container */
+    delay?:boolean;
+}
+
+interface IEntries{
+    /* type of entry, heading or entry */
+    type:string;
+    /* data per line */
+    entry:IEntryProps|IHeadingEntryProps&IAdditionalProps;
+}
+
+export interface INutritionFactProps{
+    entries:IEntries[];
+    editMode?:boolean;
+}
+
+export const NutritionFact:React.FC<INutritionFactProps>=({entries,editMode}):React.ReactElement=>
+{
+    const mainContainerRef=useRef<HTMLDivElement>(null)
+
+    const containerRef=useRef<HTMLDivElement>(null)
+
+    const delayLabel=useRef<string[]>([])
+
+    const infoRef=useRef<IInfo|null>(null)
+
+    /**
+     * renders an entry of type entry
+     * @param entry {IEntryProps} - the entry to render
+     * @returns {React.ReactElement} the rendered entry
+     */
+    const renderEntry=useCallback((entry:IEntryProps):React.ReactElement=>{
+        const {label,...rest}=entry
+        return <Entry key={label} label={label} margin='0 0 2px' padding='0 0 1px' editMode={editMode} {...rest} />
+    },[editMode])
+
+    /**
+     * renders an entry of type heading
+     * @param entry {IHeadingEntryProps} - the entry to render
+     * @returns {React.ReactElement} the rendered entry
+     */
+    const renderHeadingEntry=useCallback((entry:IHeadingEntryProps&IAdditionalProps):React.ReactElement=>{
+        const {label,ref,delay,...rest}=entry
+        if(ref) return <HeadingEntry key={label} label={label} ref={containerRef} editMode={editMode} {...rest} />
+        if(delay) {
+            delayLabel.current.push(label)
+            return <HeadingEntry key={label} label='' infoRef={infoRef} editMode={editMode} {...rest} />
+        }
+        return <HeadingEntry key={label} label={label} editMode={editMode} {...rest} />
+    }
+    ,[])
+
+    /**
+     * this sets max width for the main container of the component and also sets the content for the delayed label
+     */
+    useEffect(()=>{
```

Comment:
Why not useLayoutEffect?

Reply:
I tried useLayoutEffect but then the content of the heading for daily value explanation doesn't appear.